src/main/java/com/linkedin/xinfra/monitor/services/MultiClusterTopicManagementService.java

```diff
@@ -415,90 +414,103 @@ int numPartitions() throws InterruptedException, ExecutionException {
-            && Utils.ongoingPartitionReassignments(_adminClient).isEmpty()) {
-          LOGGER.info("{} will reassign partitions of the topic {} in cluster.", this.getClass().toString(), _topic);
-          reassignPartitions(_adminClient, brokers, _topic, partitionInfoList.size(), expectedReplicationFactor);
+      for (Object key : currentProperties.keySet()) {
+        expectedProperties.put(key, currentProperties.get(key));
+      }
+      for (Object key : _topicProperties.keySet()) {
+        expectedProperties.put(key, _topicProperties.get(key));
+      }
 
-          partitionReassigned = true;
+      if (!currentProperties.equals(expectedProperties)) {
+        LOGGER.info("MultiClusterTopicManagementService will overwrite properties of the topic {} "
+            + "in cluster from {} to {}.", _topic, currentProperties, expectedProperties);
+        Map<ConfigResource, Collection<AlterConfigOp>> newConfigs = new HashMap<>();
+        for (Map.Entry<Object, Object> entry : expectedProperties.entrySet()) {
+          newConfigs.put((ConfigResource) entry.getKey(), (Collection) entry.getValue());
         }
+        _adminClient.incrementalAlterConfigs(newConfigs);
+      }
+
+      if (partitionInfoList.size() >= brokers.size() && someBrokerNotPreferredLeader(partitionInfoList, brokers)
+          && Utils.ongoingPartitionReassignments(_adminClient).isEmpty()) {
+        LOGGER.info("{} will reassign partitions of the topic {} in cluster.", this.getClass().toString(), _topic);
+        reassignPartitions(_adminClient, brokers, _topic, partitionInfoList.size(), expectedReplicationFactor);
+        partitionReassigned = true;
+      }
 
-        if (partitionInfoList.size() >= brokers.size() && someBrokerNotElectedLeader(partitionInfoList, brokers)) {
-          if (!partitionReassigned || Utils.ongoingPartitionReassignments(_adminClient).isEmpty()) {
-            LOGGER.info("MultiClusterTopicManagementService will trigger preferred leader election for the topic {} in "
-                + "cluster.", _topic);
-            triggerPreferredLeaderElection(partitionInfoList, _topic);
-            _preferredLeaderElectionRequested = false;
-          } else {
-            _preferredLeaderElectionRequested = true;
+      if (partitionInfoList.size() >= brokers.size() && someBrokerNotElectedLeader(partitionInfoList, brokers)) {
+        if (!partitionReassigned || Utils.ongoingPartitionReassignments(_adminClient).isEmpty()) {
+          LOGGER.info("MultiClusterTopicManagementService will trigger preferred leader election for the topic {} in "
+              + "cluster.", _topic);
+          triggerPreferredLeaderElection(partitionInfoList, _topic);
+          _preferredLeaderElectionRequested = false;
+        } else {
+          _preferredLeaderElectionRequested = true;
+
+          if (partitionInfoList.size() >= brokers.size() && someBrokerNotElectedLeader(partitionInfoList, brokers)) {
```

Comment:
I find it hard to understand the logic here.
Isn't partitionInfoList.size() >= brokers.size() && someBrokerNotElectedLeader(partitionInfoList, brokers)
guaranteed to be true, because of the outer condition?

There seems to be quite some code duplication between the inner conditions and the outer conditions.